src/devices/srsmith_pool_srs_2c_tx.c

```diff
@@ -0,0 +1,175 @@
+*/
+
+static int srsmith_pool_srs_2c_tx_callback(r_device *decoder, bitbuffer_t *bitbuffer)
+{
+    data_t *data;
+
+    if (bitbuffer->num_rows != 1)
+        return DECODE_ABORT_EARLY;
+
+    if (bitbuffer->bits_per_row[0] != 144)
+        return DECODE_ABORT_LENGTH;
+
+    // this is heavily derived from Maverick XR30 BBQ Sensor
+    uint8_t b[18];
+
+    // check for correct preamble/sync (0xaaaaaad391d391)
+    if (bitbuffer->bb[0][0] == 0x55) {
+        bitbuffer_extract_bytes(bitbuffer, 0, 7, b, 18 * 8); // shift in case first bit was not received properly
+    }
+    else if (bitbuffer->bb[0][0] == 0xaa) {
+        bitbuffer_extract_bytes(bitbuffer, 0, 8, b, 18 * 8); // shift in case first bit was received properly
+    }
+    else {
+        return DECODE_ABORT_EARLY; // preamble/sync missing
+    }
+    if (b[0] != 0xaa || b[1] != 0xaa || b[2] != 0xaa || b[3] != 0xd3 || b[4] != 0x91 || b[5] != 0xd3 || b[6] != 0x91)
+        return DECODE_ABORT_EARLY; // preamble/sync missing
+
+    int totallength = bitbuffer->bits_per_row[0];
+    int packetsize  = (int)b[7];
+
+    // packet actually starts at b[8]
+    uint32_t unknown_field = ((uint32_t)b[8] << 24) | (b[9] << 16) | (b[10] << 8) | (b[11]); // not sure what these four bytes are.
+
+    // get xmitted pin (byte 12, inverted, reversed, bottom nibble)
+    uint8_t pin_container          = b[12];
+    uint8_t inverted_pin_container = ~pin_container;                   // invert the pin
+    uint8_t reversed_pin           = reverse8(inverted_pin_container); // reverse the bits
+
+    // convert just the first four bits to binary string
+    char pin_string[5] = {'\0'};
+    snprintf(pin_string, sizeof(pin_string), "%d%d%d%d",
+            (reversed_pin & 0x80 ? 1 : 0),
+            (reversed_pin & 0x40 ? 1 : 0),
+            (reversed_pin & 0x20 ? 1 : 0),
+            (reversed_pin & 0x10 ? 1 : 0));
+
+    // button that was pushed
+    uint8_t button_id = b[13];
+
+    // get label
+    char button_string[50] = {'\0'};
+
+    switch (button_id) {
+    case BUTTON_ID_ONE:
+        snprintf(button_string, sizeof(button_string), "On/Off Channel 1");
+        break;
+    case BUTTON_ID_TWO:
+        snprintf(button_string, sizeof(button_string), "On/Off Channel 2");
+        break;
+    case BUTTON_ID_S:
+        snprintf(button_string, sizeof(button_string), "Color Sync");
+        break;
+    case BUTTON_ID_M:
+        snprintf(button_string, sizeof(button_string), "ON/OFF Control - M");
+        break;
+    default:
+        snprintf(button_string, sizeof(button_string), "Unknown");
+        break;
+    }
+
+    // grab CRCs
+    // commenting out the below two lines because we're not using this crc
+    // see above for description.
+    //
+    // however, i'd still like the code to be here in case we need it in the future.
+    // uint8_t baby_crc            = b[14];
+    // uint8_t baby_calculated_crc = crc8(&b[8], 6, 1, 1);
```

Comment:
Is the CRC poly really `1`? That's parity not CRC.

Reply:
renamed in code.